app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,12 @@
+class ShoppingListController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @recipes = current_user.recipes
+    @foods = Food.joins(:recipe_foods).where(user_id: current_user.id)
+    @total_price = 0
+    @recipes.each do |recipe|
+      @total_price += recipe.total_price
+    end
+  end
+end
```

Comment:
- [ ] We don't implement SQL Queries inside the controller, we use the model to keep the SQL queries implementation.

Reply:
Thank you. It is done. 💪